tests/by-util/test_install.rs

```diff
@@ -1613,3 +1614,55 @@ fn test_target_file_ends_with_slash() {
         .fails()
         .stderr_contains("failed to access 'dir/target_file/': Not a directory");
 }
+
+#[test]
+// Matches part of tests/install/install-C-root.sh
+fn test_install_root() {
+    let ts = TestScenario::new(util_name!());
+    let at = ts.fixtures.clone();
+    at.touch("a");
+
+    let run_and_check = |args: &[&str], expected_uid: u32, expected_gid: u32| {
+        if let Ok(result) = run_ucmd_as_root(&ts, args) {
+            result.success();
+            assert!(at.file_exists("b"));
+            assert!(result.stdout_str().contains("'a' -> 'b'\n"));
+
+            // Check the UID and GID of the file
+            let metadata = fs::metadata(at.plus("b")).unwrap();
+            assert_eq!(metadata.uid(), expected_uid);
+            assert_eq!(metadata.gid(), expected_gid);
+        } else {
+            print!("Test skipped; requires root user");
+        }
+    };
+
+    run_and_check(&["-Cv", "-o1", "-g1", "a", "b"], 1, 1);
+    run_and_check(&["-Cv", "-o2", "-g1", "a", "b"], 2, 1);
+    run_and_check(&["-Cv", "-o2", "-g2", "a", "b"], 2, 2);
+}
+
+#[test]
+fn test_install_as_root() {
```

Comment:
I think the two test functions can be merged. They have almost the same names (`test_install_root` vs. `test_install_as_root`) and the code looks almost identical.